### DISCUSSION TITLE
How does "Execute command" work?

### DISCUSSION THREAD

**Discussion by plitsas** [epoptes/epoptes#105] [Q&A]
Hello,
I have epoptes server on teacher PC setup and runnign ok.
I have a mixed lab with ubuntu and a custom debian distro.
From the server I see all the machines and I can do to almost everything from epoptes server window:
eg 
-show teacher screen everywhere
-send an alertbox message
-Shutdown/reboot/lock/unlock PCs
-I can open consoles locally on server and client
-control sound

My problem is that **I can't run "execute custom command" on the custom debian clients** (on ubuntu clients they get executed ok).
Also, something that might help is that from server window (where I can see icons of all the connected clients) I can see screenshots from the ubuntu PCs but **I don't see screenshots of the custom debian PCs**.

I don't know if you can pinpoint what the problem is from my info.
If you can't, can you please, tell me the steps of the "execute custom command" workflow so I can try and debug it myself? The custom debian has many useful stuff I use so it is no option to abandon it.

Thanks in advance

**Comment by alkisg** [comment]
On the custom debian, open a normal terminal, and run /usr/sbin/epoptes-client.

See if that makes things work. If yes, then make sure it runs on user login; normally it's run by /etc/xdg/autostart.

**Comment by plitsas** [comment]
Thanks for the quick reply.
I run epoptes-client with a script from user's StartUp folder. 
epoptes-client runs ok. 
The only 2 things that doesn't work are "execute command on client" and I don't see the screenshot thumbnails of the client PCs.

I should note that I have this problem in both 32bit and 64bit version of this custom debian. It is probably the OS fault. If I you can tell me the steps of the execute command workflow I might be able to debug and see what is the problem.

**Comment by alkisg** [comment]
OK, please do what I said and paste the terminal output.
There should be two epoptes-client processes running, and you're missing the user one.

**Comment by plitsas** [comment]
ok. 
Thanks I'll do that and post results here.
It will take a while since I;ll be on this lab on Monday
Thanks again

**Comment by plitsas** [comment]
Thanks. Your suggestion helped me locate the problem.
The problem was that I was logged in as root in the clients . I logged with a normal user started again the 2 epoptes-clients processes (as root & as simple user) and it worked.
My problem now is that there is no option for me to re-setup everything for simple user account. 

Is it possible to make epoptes-client work while the clients PCs are logged as root ? 
Thanks again for your time


(edit)
Managed to do it:
I start from root 2 proecesses:
`/usr/sbin/epoptes-client`
and
```
xhost +
su - user1 -c "/usr/sbin/epoptes-client"
```

Thanks for your help :)

**Comment by alkisg** [comment]
Hi plitsas,

sorry, running a desktop environment as root is considered a security risk and many programs, including Epoptes, don't officially support it.

It's good to hear that you found a workaround by running just Epoptes as a user, though. :)

**Comment by plitsas** [comment]
I know it is a risk. 
As I said it is custom OS and everything is executed as a user or using firejail.
Thanks again
